File: Pencil.java

public class Pencil extends Item {


	Pencil(String name,double price , String description , int stock , int id , double tipSize ,PENCIL_TYPE pencilType) throws IllegalCharacterException{

		super(name , price ,description , stock , id);  	//calls Item constructor
		
		this.tipSize = tipSize ;
		
		this.pencilType = pencilType ;

	}


    //Variables
	
	private double tipSize;
	
	public enum PENCIL_TYPE {
		
		H,B,HB ; 
		
	}

	private PENCIL_TYPE pencilType ;


	//Methods
	
//---------------------getDetails implementation-------------------------
	
	public String getDetails() {
		
	
		return ", Type: " + pencilType.toString() +", Tip Size: " + Double.toString(tipSize) ;
		
	}
	
	
//--------------------------Setters--------------------------------------
	
	
	public void setPencilType(PENCIL_TYPE size) {  // pencil type
		
		this.pencilType = size ;
		
	}

	public void setTipSize(double size){   		//tip size

		this.tipSize = size ;

	}
	
	
//-------------------------Getters---------------------------------------
	
	
	public double getTipSize() {    				//tip size
		
		return tipSize ;
		
	}

	public PENCIL_TYPE getPencilType() {		//pencil type

		return this.pencilType ;
		
    }
    
}

File: C++/Pencil.cpp
#include "Pencil.h"

Pencil::Pencil(string n, double p, string d, int s, int i, double t, PENCIL_TYPE p_t): Item(n, p, d, s, i), tipSize(t), pencilType(p_t) {

    this->enum_to_string[0] = "H";
    this->enum_to_string[1] = "B";
    this->enum_to_string[2] = "HB";

}

string Pencil::getDetails() {

    ostringstream detailsStream;
    detailsStream << fixed << setprecision(2) << ", Type: " << this->enum_to_string[this->pencilType] << ", Tip Size: " << this->tipSize << "mm";

    return detailsStream.str();
        
}

string Pencil::getCategory() {
    return "pencil";
}

void Pencil::setTipSize(double s) {
    this->tipSize = s;
}

double Pencil::getTipSize() {
    return this->tipSize;
}

void Pencil::setPencilType(PENCIL_TYPE type) {
    this->pencilType = type;
}

Pencil::PENCIL_TYPE Pencil::getPencilType() {
    return this->pencilType;
}

string Pencil::getPencilTypeString() {

    return this->enum_to_string[this->pencilType];

}
